### PR TITLE
Fedora Build Fix + Fedora Build Instructions

### PR DESCRIPTION
**Pull request by Yiannis128**
I've fixed the build process for Fedora, and now it builds like it does with Ubuntu LTS 20.04

### CODE REVIEW

File: src/esbmc/CMakeLists.txt [hunk 1]
```diff
@@ -45,6 +45,6 @@ message("Looking at ${CVC_LINK_DIR}")
 target_link_libraries(esbmc ${OLD_FRONTEND_TARGETS} ${SOLIDITY_FRONTEND_TARGETS} ${PYTHON_FRONTEND_TARGETS}
                             ${GOTO_CONTRACTOR_TARGETS} ${JIMPLE_FRONTEND_TARGETS} ${JIMPLE_FRONTEND_TARGETS}
                             clangcfrontend clangcppfrontend filesystem symex pointeranalysis langapi util_esbmc bigint
-                            solvers clibs gotoalgorithms cache ${Boost_LIBRARIES} goto2c)
+                            solvers clibs gotoalgorithms cache ${Boost_LIBRARIES} goto2c m stdc++)
```
Comment (rafaelsamenezes) [inline src/esbmc/CMakeLists.txt]: libm should only be added for Unix targets.

**Review by rafaelsamenezes — CHANGES_REQUESTED** [review]
Overall its ok. Just make libm unavailable for windows targets.

**Comment by Yiannis128** [comment]
> Overall its ok. Just make libm unavailable for windows targets.

Is the library unavailable on Windows? Or is it another reason?

File: CMakeLists.txt [hunk 2]
```diff
@@ -57,7 +57,6 @@ include(Utils)
 include(OSConfiguration)
 include(Options)
 include(SendFileHack)
-include(AppleConfiguration)
```
Comment (Yiannis128) [inline CMakeLists.txt:60]: @rafaelsamenezes AppleConfiguration is included in OSConfiguration already, so I've removed it here. Is that OK?

File: src/esbmc/CMakeLists.txt (hunk 1 above)
Comment (Yiannis128) [reply src/esbmc/CMakeLists.txt]: OK, I've created a new variable that gets set differently for each OS in OSConfiguration. It's called `OS_INCLUDE_LIBS`, it gets used on targets that did not compile successfully. It's set as a blank value for any OS other than UNIX now.

**Review by lucasccordeiro — APPROVED** [review]

**Review by rafaelsamenezes — APPROVED** [review]

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @Yiannis128.

**Pull request merged**